static/analysis.go

```diff
@@ -40,6 +40,16 @@ func analyzeApplication(prog *ssa.Program, typ types.Type) configkit.Application
 					),
 				)
 			}
+		case "RegisterProcess":
```

Comment:
I wonder if we want to try to remove a bit of this duplication here since the code is so similar. We don't *have* to, I don't think the duplication is a big deal, but just asking the question.

For example, we could have a function that works something like `analyzeHandlerFromArguments prog, args,  configkit.ProcessHandlerType)`.

Reply:
I believe that would require `if mi, ok := args[0].(*ssa.MakeInterface)` check to be completed inside `analyzeHandlerFromArguments()` function. This makes the result of this function indeterministic and should probably introduce boolean `ok` to indicate that the check above fails. At the end we will have a call of this function looking something like: 
```golang 

if h, ok := analyzeHandlerFromArguments(prog, args, configkit.ProcessHandlerType); ok {
  app.HandlersValue.Add(h)
}
```

I am not sure if we are winning much by doing this.

Reply:
maybe it could just take the `HandlerSet` and we could call it something like `addHandlerFromArguments()`?

Reply:
let me try that

Reply:
Ok, that should be addressed in https://github.com/dogmatiq/configkit/pull/130/commits/860b0a6ba50166e5803e5c5918140b53f50f38e2.